subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/inputs/undeclared/UndeclaredBuildInputsIntegrationTest.groovy

```diff
@@ -592,6 +602,57 @@ class UndeclaredBuildInputsIntegrationTest extends AbstractConfigurationCacheInt
         outputContains("value = value")
     }
 
+    @Issue("https://github.com/gradle/gradle/issues/25044")
+    @Requires(UnitTestPreconditions.Jdk11OrLater)
+    def "plugin can read file within jar"() {
+        def testFile = JarUtils.jar(testDir.file("thing.jar")) {
+            manifest {}
+
+            entry("foo.txt", "bar")
+        }
+
+        buildKotlinFile << """
+            import ${DefaultTask.name}
+            import ${Project.name}
+            import ${Plugin.name}
+            import ${TaskAction.name}
+            import ${File.name}
+            import ${URI.name}
+            import ${FileSystems.name}
+            import ${Files.name}
+
+            class SomePlugin : Plugin<Project> {
+                override fun apply(project: Project) {
+                    project.tasks.register("readFileWithinJarTask", ReadFileWithinJarTask::class.java)
+                }
+            }
+
+            abstract class ReadFileWithinJarTask : DefaultTask() {
+                @TaskAction
+                fun run() {
+                    try {
```

Comment:
Nitpicking: I would lose the try/catch statement, as Groovy does not enforce handling of checked exceptions.

Reply:
done with thanks!